File: 1022. Sum of Root To Leaf Binary Numbers/_1022.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def sumRootToLeaf(self, root: TreeNode) -> int:
        res = list()
        if root:
            self.getValue(root, root.val, res)
        return sum(res)

    def getValue(self, root, value, res):
        if not root.left and not root.right:
            res.append(value)
            return
        value *= 2
        if root.left:
            self.getValue(root.left, value+root.left.val, res)
        if root.right:
            self.getValue(root.right, value+root.right.val, res)

File: 1022. Sum of Root To Leaf Binary Numbers/1022.go
package leetcode

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
type TreeNode struct {
	Val int
	Left *TreeNode
	Right *TreeNode
}

 func sumRootToLeaf(root *TreeNode) int {
    res := make([]int, 0)
	if root!=nil{
		getValue(root, root.Val, &res)
	}

	count := 0
	for _, v := range res{
		count += v
	}
	return count
}

func getValue(root *TreeNode, value int, res *[]int){
	if root.Left == nil && root.Right == nil{
		*res = append(*res, value)
		return 
	}

	value *= 2
	if root.Left != nil{
		getValue(root.Left, value+root.Left.Val, res)
	}
	if root.Right != nil{
		getValue(root.Right, value+root.Right.Val, res)
	}
}